src/functions-and-arrays.js

```diff
@@ -1,20 +1,49 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
-
+function maxOfTwoNumbers(number1, number2) {
+  let maxNumber = number1 > number2;
+  if (maxNumber) {
+    return number1;
+  }
+  
+  return number2;
+  }
 
+//console.log(maxOfTwoNumbers(5,6))
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
+function findLongestWord(array) {
+  let longestWord = 'a';
+  if (array.length === 0) {
+    longestWord = null;
+  }
+
+  for (let i = 0; i < array.length; i++) {
+    
+    if (array[i].length > longestWord.length) {
+      longestWord = array[i];
+    }
+  }
+
+  return longestWord;
+}
 
+//console.log(findLongestWord(words));
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbersArray) {
+  let sumNumbersArray = 0;
+  for (i = 0; i < numbersArray.length; i++) {
+    sumNumbersArray += numbersArray[i]; 
+  }
+  return sumNumbersArray
+}
 
+//console.log(sumNumbers(numbers));
 
 
 // Iteration #3.1 Bonus:
```

Comment:
const mixedArr = [6, 12, "miami", 1, true, "barca", "200", "lisboa", 8, 10];

function sum(mixedArr) {
  if (mixedArr.length === 0) return 0;

  let result = 0;
  for (let i = 0; i < mixedArr.length; i++) {
    const element = mixedArr[i];

    if (typeof element === "string") {
      result += element.length;
    } else if (element === true) {
      result += 1;
    } else if (typeof element === "object") {
      throw new Error("Unsupported data type sir or ma'am");
    } else {
      result += mixedArr[i];
    }
  }
  return result;
}